fmf/utils.py

```diff
@@ -304,13 +306,18 @@ def check_clause(clause):
     if not isinstance(data, dict):
         raise FilterError("Invalid data type '{0}'".format(type(data)))
 
-    # Make sure that data dictionary contains lists of strings
+    # Add fmf name to the data dictionary if provided
     data = copy.deepcopy(data)
+    if name is not None:
```

Comment:
Could we use same "name" for attribute and injected data? eg. 'fmf-name' for both.